app/core/store/modules/announcements.js

```diff
@@ -0,0 +1,123 @@
+const fetchJson = require('../../api/fetch-json')
+import { getNew, getList, read } from '../../api/announcements'
+
+export default {
+  namespaced: true,
+  state: {
+    announcementInterval: false,
+    announcementModalOpen: false,
+    announcements: [],
+    unread: 0,
+    display: {},
+    moreAnnouncements: true,
+    showMoreAnnouncementButton: false
+  },
+  mutations: {
+    setUnread (state, unread) {
+      state.unread = unread
+    },
+    setAnnouncementInterval (state, interval) {
+      state.announcementInterval = interval
+    },
+    setAnnouncementModalOpen (state, mode) {
+      state.announcementModalOpen = mode
+    },
+    setAnnouncements (state, {anns, append}) {
+      if(append) {
+        anns.forEach((ann) => {
+          if(!_.find(state.announcements, {_id: ann._id})) {
+            state.announcements.push(ann)
+          }
+        })
+      }
+      else
+        state.announcements = [...anns]
+    },
+    setDisplay (state, ann) {
+      state.display = ann
+    },
+    readAnnouncement (state, id) {
+      state.announcements = state.announcements.map((ann) => {
+        if(ann._id === id) {
+          if(!ann.read) {
+            state.unread -= 1;
+          }
+          ann.read = true
+        }
+        return ann
+      })
+    },
+    setMoreAnnouncement (state, more) {
+      state.moreAnnouncements = more
+    },
+    setShowMoreButton (state, more) {
+      state.showMoreAnnouncementButton = more
+    }
+  },
+  actions: {
+    openAnnouncementModal ({ commit }, announcement) {
+      commit('setDisplay', announcement)
+      commit('setAnnouncementModalOpen', true)
+    },
+    closeAnnouncementModal ({commit}) {
+      commit('setAnnouncementModalOpen', false)
+    },
+    startInterval ({commit, dispatch}, fromNav) {
+      let interval = setInterval(() => { dispatch('checkAnnouncements', fromNav) }, 600000) // every 10 mins
+      commit('setAnnouncementInterval', interval)
+    },
+    checkAnnouncements ({ commit }, fromNav) {
+      if(me.isAnonymous()) {
+        return
+      }
```

Comment:
Are we able to show homepage banner announcements to anonymous users?

Reply:
okay